storage/src/test/java/org/apache/kafka/server/log/remote/storage/LocalTieredStorage.java

```diff
@@ -254,9 +255,10 @@ public void configure(Map<String, ?> configs) {
 
         if (transfererClass != null) {
             try {
-                transferer = (Transferer) getClass().getClassLoader().loadClass(transfererClass).newInstance();
-
-            } catch (InstantiationException | IllegalAccessException | ClassNotFoundException | ClassCastException e) {
+                transferer = (Transferer) getClass().getClassLoader().loadClass(transfererClass)
+                        .getDeclaredConstructor().newInstance();
```

Comment:
`clazz.newInstance()` has been deprecated since Java 9. 
The documentation recommends using `clazz.getDeclaredConstructor().newInstance()` instead.”